specutils/spectra/spectrum1d.py

```diff
@@ -442,6 +442,81 @@ def _spectral_slice(self, item):
         reg = SpectralRegion(start, stop)
         return extract_region(self, reg)
 
+    def collapse(self, method, axis=None, physical_type=None):
+        """
+        Collapse the flux array given a method. Will collapse either to a single
+        value (default), over a specified numerical axis or axes if specified, or
+        over the spectral or non-spectral axes if ``physical_type`` is specified.
+
+        If the collapse leaves the spectral axis unchanged, a `~specutils.Spectrum1D`
+        will be returned. Otherwise an `~astropy.units.Quantity` array will be
+        returned.
+
+        Note that these calculations are not currently uncertainty-aware, but do
+        respect masks.
+
+        Parameters
+        ----------
+
+        method : str
+            The method by which the flux will be collapsed. Options are 'mean',
+            'min', 'max', 'sum', and 'median'.
+        axis : int, tuple, optional
+            The axis or axes over which to collapse the flux array.
+        physical_type : str, optional
+            Either 'spectral' or 'spatial'.
+
+        Returns
+        -------
+        :class:`~specutils.Spectrum1D` or :class:`~astropy.units.Quantity`
+
+        """
+        collapse_func = {"mean": np.nanmean, "max": np.nanmax, "min": np.nanmin,
+                         "median": np.nanmedian, "sum": np.nansum}
+
+        if physical_type is not None and axis is not None:
+            raise ValueError("Cannot set both axis and physical_type inputs")
+
+        if physical_type == 'spectral':
+            axis = -1
+        elif physical_type == 'spatial':
+            # generate tuple if needed for multiple spatial axes
+            axis = tuple([x for x in range(len(self.flux.shape) - 1)])
+        else:
```

Comment:
This doesn't seem to handle the case where `physical_type` is `None` and `axis` is not set? Perhaps by default, we assume that the default collapse axis is spectral.

Reply:
I assumed that the default case with neither set is to collapse the entire thing to a single value.